cli/azd/cmd/login.go

```diff
@@ -93,7 +93,7 @@ func (la *loginAction) SetupFlags(persistent *pflag.FlagSet, local *pflag.FlagSe
 func ensureLoggedIn(ctx context.Context) error {
 	azCli := commands.GetAzCliFromContext(ctx)
 	_, err := azCli.GetAccessToken(ctx)
-	if errors.Is(err, tools.ErrAzCliNotLoggedIn) {
+	if errors.Is(err, tools.ErrAzCliNotLoggedIn) || errors.Is(err, tools.ErrAzCliRefreshTokenExpired) {
 		if err := runLogin(ctx, false); err != nil {
```

Comment:
As a further improvement, I think `runLogin` should print a message when an interatcive login is required AND console output is not disabled. I'm not sure if we currently have a function that checks "should stdout output be allowed?" Is there existing code here?

Reply:
I suspect there would be two cases:
1. In CI builds (is this handled with `no-prompt`?), we want anything interactive to correctly fail, so we want to ensure the message is not printed out (likely the CI check would happen prior)
2. As you expect, the output selection likely means that we need to suppress any info messages, and only the result should be printed.